include/stream_list_priv.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * stream_list_priv.h
+ *
+ * list of SRTP streams, keyed by SSRC
+ *
+ * Alba Mendez
+ */
+/*
+ *
+ * Copyright (c) 2001-2017, Cisco Systems, Inc.
+ * Copyright (c) 2022, Dolby Laboratories, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ *   Redistributions in binary form must reproduce the above
+ *   copyright notice, this list of conditions and the following
+ *   disclaimer in the documentation and/or other materials provided
+ *   with the distribution.
+ *
+ *   Neither the name of the Cisco Systems, Inc. nor the names of its
+ *   contributors may be used to endorse or promote products derived
+ *   from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef SRTP_STREAM_LIST_PRIV_H
+#define SRTP_STREAM_LIST_PRIV_H
+
+#include <srtp_priv.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * srtp_stream_list_t holds a list of srtp_stream_t, each identified
+ * by their SSRC.
+ *
+ * the API was extracted to allow downstreams to override its
+ * implementation by defining the `SRTP_NO_STREAM_LIST` preprocessor
+ * directive, which removes the default implementation of these
+ * functions. if this is done, the `next` & `prev` fields are free for
+ * the implementation to use.
```

Comment:
In my mind that would be a natural progression of this PR, not necessarily the purpose of this PR. We can consider going the full steep of making stream & stream list 100% independent of each other now if that is preferred. Will be a bit more work but will result in a clean change in the end.

Reply:
removing `next` and `prev` from `srtp_stream_ctx_t` would mean the default impl should have to change from being an intrusive linked list to a non-intrusive one, with the performance implications that come with it. so I agree with @pabuhler that this can be done later. we can remove the "are free for the implementation to use" bit if you want